### ISSUE TITLE
Disable to detect relationship of tables in metadata

### ISSUE THREAD

**Issue by XiaPoppy** [sdv-dev/SDV#1870]
### Environment Details

* SDV version: 1.10.0
* Python version: 3.11.0
* Operating System: M1 Mac

### Error Description

I tried to detect metadata from csv files from open datasource [(NBA Database)](https://www.kaggle.com/datasets/wyattowalsh/basketball?select=csv). But I got result that they are totally disjoint.

### Steps to reproduce
download csv files and store them under folder csv
```
meta_data = MultiTableMetadata()
meta_data = meta_data.detect_from_csvs(folder_name='csv/')
```
and with warning output: 
`UserWarning: Could not automatically add relationships for all tables. The relationships in the dataset are disjointed. Tables ['play_by_play', 'game_info', 'player', 'other_stats', 'officials', 'game_summary', 'draft_combine_stats', 'line_score', 'team', 'game'] are not connected to any of the other tables.`

**Comment by srinify** [comment]
Hi @XiaPoppy I dug into this a bit further.

To use the multi-table synthesizers in SDV, you need your data tables to have a few things:

1. **One-to-many relationships**: Your tables currently need to have a one-to-many (or one-to-one) relationship where:

- one table is the `parent` and has a primary key
- the other table is the `child` and has a foreign key 

2. **Referential integrity**:  SDV also expects all values in the foreign key column of the child table to match a value in the primary key column of the parent table (also known as referential integrity). If you want to exclude values that don't meet this criteria, we've added a [helpful utility function](https://docs.sdv.dev/sdv/multi-table-data/data-preparation/loading-data#drop_unknown_references) to drop those rows.

**Issues with the NBA Datasets from Kaggle**

The values in the primary key column have to be unique (one per row) and I found that this wasn't the case for some of the tables. For example, the `game_info` table has a few rows with the same `game_id` value.

<img width="435" alt="Screenshot 2024-04-04 at 5 49 06 AM" src="https://github.com/sdv-dev/SDV/assets/801507/f28d3872-70b8-4122-a03b-53fe83937505">

You have a few potential ways forward here:

- If you're just trying to try SDV with some public datasets, I'd either spend some time cleaning up this NBA dataset so that the conditions for SDV are met or switching to one of our demo datasets: https://docs.sdv.dev/sdv/multi-table-data/data-preparation/loading-data#demo-data
- It's pretty common to load data from relational databases, where most organizations store their data, into SDV for generating synthetic day. Most databases enforce things like unique keys for primary key columns and referential integrity between tables. So I'd recommend trying SDV a spin using data from your database if you have one!

**Comment by XiaPoppy** [comment]
Thanks for your reply. And I have the following questions about the relation detection:

it can only detect 1-1 and 1-many tables, but totally non-possible to detect many-many tables?

**Comment by srinify** [comment]
@XiaPoppy correct! Currently SDV's multi-table capabilities don't support many-to-many tables

**Comment by srinify** [comment]
I'm closing this issue for now because it feels resolved. If that's not the case, please feel free to re-open :)
